From 3dfd924741a7c528429628746476cfdd656d4891 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 22 Nov 2021 20:23:27 +0200
Subject: [PATCH] [SW-36216] gaudi2: security: update pcie lbw rrs

As per the updated lbw rr spec, pcie elbi and pcie mstr rrs need to be
configured apart from the existing pcie lbw rrs.

Existing : pcie lbw rrs
Update : pcie lbw + elbi + mstr rrs.

As pcie elbi, lbw and mstr blocks are consecutive, the iterative
framework has been updated to configure all 3 blocks.

Change-Id: I7c3aa67dfe98b743db98914799db523349d6a99b
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 3dc07c9..5d76eff 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -19,7 +19,7 @@
 #include <psoc_arc1_mstr_if_core_hbw_regs.h>
 #include <psoc_arc1_mstr_if_core_lbw_regs.h>
 #include <dcore0_rtr0_mstr_if_rr_shrd_lbw_regs.h>
-#include <pcie_lbw_rr_mstr_if_rr_shrd_lbw_regs.h>
+#include <pcie_elbi_rr_mstr_if_rr_shrd_lbw_regs.h>
 #include <jt_mstr_if_rr_shrd_lbw_regs.h>
 #include <sft0_lbw_rtr_if_mstr_if_rr_shrd_lbw_regs.h>
 #include <pcie_mstr_rr_mstr_if_rr_shrd_hbw_regs.h>
@@ -182,11 +182,13 @@
 #define DCORE_DIST	0x200000
 #define SFT_DIST	0x10000
 #define IF_RTR_DIST	0x8000
-#define PCIE_INST_NUM	1
+#define PCIE_INST_NUM	3 /* ELBI, LBW, MSTR */
+#define PCIE_INST_DIST	0x1000
 #define PSOC_INST_NUM	3 /* JTAG, SMI, I2C */
 #define PSOC_INST_DIST	0x1000
-#define SFT_INST_NUM	PCIE_INST_NUM
+#define SFT_INST_NUM	1
 #define SHIFT_4K_POS	12
+#define PCIE_HBW_NUM	1
 
 static struct hl_sec_rrs_cfg hl_sec_lbw_rrs_tbl[] = {
 	[HL_SEC_ENGINE_IF_RTR] = {
@@ -201,13 +203,13 @@
 	},
 	[HL_SEC_ENGINE_PCIE] = {
 		.instances_num = PCIE_INST_NUM,
-		.instances_dist = 0,
-		.min_short_breg = mmPCIE_LBW_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
-		.max_short_breg = mmPCIE_LBW_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
-		.min_full_breg = mmPCIE_LBW_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
-		.max_full_breg = mmPCIE_LBW_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
-		.hit_wr_breg = mmPCIE_LBW_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmPCIE_LBW_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
+		.instances_dist = PCIE_INST_DIST,
+		.min_short_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
+		.max_short_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
+		.min_full_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
+		.max_full_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
+		.hit_wr_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
+		.hit_rd_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
 	},
 	[HL_SEC_ENGINE_PSOC] = {
 		.instances_num = PSOC_INST_NUM,
@@ -250,7 +252,7 @@
 		.hit_rd_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
 	},
 	[HL_SEC_ENGINE_PCIE] = {
-		.instances_num = PCIE_INST_NUM,
+		.instances_num = PCIE_HBW_NUM,
 		.instances_dist = 0,
 		.min_short.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
 		.min_short.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
